app/_src/gateway/plugin-development/quickstart.md

```diff
@@ -0,0 +1,679 @@
+    end
+    ```
+
+    Next, let's validate our changes work as expected with automated and manual tests.
+
+1. **Test configurable `my-plugin`**
+
+    Now that `my-plugin` is configurable, let's look at how we can validate our changes.
+
+    Without changing anything we can re-run the tests we already have with `pongo run`:
+
+    ```sh
+    pongo run
+    ```
+
+    Our tests should continue to report successful:
+
+    ```sh
+    [  PASSED  ] 2 tests.
+    ```
+
+    Our test is still valid because the default value in our configurable field matches the value
+    in our test case. Let's see how we can validate a different value for the `response_header_name` field.
+
+    Modify the `setup` function inside the `spec/01-integration_spec.lua` module so that the `my-plugin` that is
+    added to the database is configured with a different value for the `response_header_name` field.
+
+    Here is the code:
+
+    ```lua
+    -- Add the custom plugin to the test route
+    blue_print.plugins:insert {
+      name = PLUGIN_NAME,
+      route = { id = test_route.id },
+      config = {
+        response_header_name = "X-CustomHeaderName",
+      },
+    }
+    ```
+
+    Use Pongo to re-run the test:
+
+    ```sh
+    pongo run
+    ```
+
+    If we've changed everything properly, our tests should fail and produce a report similar to the following:
+
+    ```sh
+    /kong-plugin/spec/my-plugin/01-integration_spec.lua:75: Expected header:
+    (string) 'X-MyPlugin'
+    But it was not found in:
+    (table: 0x484c9942b180) {
+      [Connection] = 'keep-alive'
+      [Content-Length] = '1016'
+      [Content-Type] = 'application/json'
+      [Date] = 'Mon, 18 Mar 2024 13:49:06 GMT'
+      [Server] = 'mock-upstream/1.0.0'
+      [Via] = 'kong/{{page.release}}'
+      [X-CustomHeaderName] = 'response'
+      [X-Kong-Proxy-Latency] = '1'
+      [X-Kong-Request-Id] = '92dfcf56b12b0d29f54ed9295aed6356'
+      [X-Kong-Upstream-Latency] = '2'
+      [X-Powered-By] = 'mock_upstream' }
+    stack traceback:
+            /kong-plugin/spec/my-plugin/01-integration_spec.lua:75: in function </kong-plugin/spec/my-plugin/01-integration_spec.lua:66>
+    [  FAILED  ] /kong-plugin/spec/my-plugin/01-integration_spec.lua:66: my-plugin: [#postgres] The response gets a 'X-MyPlugin' header (4.58 ms)
+    ```
+
+    The tests fail because while we updated the plugin configuration, we did not update the expected header name in the test case. To fix 
+    the tests, modify the test assertion to match our configured header name.
+
+    Change this line:
+
+    ```lua
+    local header_value = assert.response(r).has.header("X-MyPlugin")
+    ```
+
+    To expect the new configured header name:
+ 
+    ```lua
+    local header_value = assert.response(r).has.header("X-CustomHeaderName")
+    ```
+
+    When you re-run the tests:
+
+    ```sh
+    pongo run
+    ```
+
+    Pongo should report a successful test run:
+
+    ```sh
+    [  PASSED  ] 2 tests.
+    ```
+
```

Comment:
At this point I'd love to see an example where we test both the default value, and a configured value. What is our recommended way to test permutations? Copy/paste? A loop?